src/parachain/redeem.ts

```diff
@@ -153,14 +153,11 @@ export class DefaultRedeemAPI {
      * @returns A boolean value
      */
     isRequestSuccessful(events: EventRecord[]): boolean {
```

Comment:
I think eventually, these functions can be all moved into one. Since now you have typed events, it can just take the event vector plus the event you are searching for as input. What do you think? Can be done in a follow-up PR though! This one is already very large.

Reply:
That's a great idea! Something similar is already being done here: https://github.com/interlay/polkabtc-js/blob/dan/refactoring/src/parachain/redeem.ts#L133